nmagent/internal/wireserver_test.go

```diff
@@ -34,19 +34,35 @@ func TestWireserverTransportPathTransform(t *testing.T) {
 			"happy path",
 			http.MethodGet,
 			"/test/path",
-			"/machine/plugins/?comp=nmagent&type=test/path",
+			"/machine/plugins?comp=nmagent&type=test%2Fpath",
 		},
 		{
 			"empty",
 			http.MethodGet,
 			"",
-			"/machine/plugins/?comp=nmagent&type=",
+			"/machine/plugins?comp=nmagent&type=",
 		},
 		{
 			"monopath",
 			http.MethodGet,
 			"/foo",
-			"/machine/plugins/?comp=nmagent&type=foo",
+			"/machine/plugins?comp=nmagent&type=foo",
+		},
+		{
+			"publish",
+			http.MethodPost,
+			//nolint:lll // not a useful linter in a test
+			"/NetworkManagement/interfaces/10.224.0.33/networkContainers/99e8b0e0-3524-490b-b2ca-31928e6d75a3?authenticationToken=6e4f70a6-a379-4f76-8e7a-2ae18392dc07&api-version=1",
+			//nolint:lll // not a useful linter in a test
+			"/machine/plugins?comp=nmagent&type=NetworkManagement%2Finterfaces%2F10.224.0.33%2FnetworkContainers%2F99e8b0e0-3524-490b-b2ca-31928e6d75a3%2FauthenticationToken%2F6e4f70a6-a379-4f76-8e7a-2ae18392dc07%2Fapi-version%2F1",
+		},
+		{
+			"publish with munged uri",
+			http.MethodPost,
+			//nolint:lll // not a useful linter in a test
+			"/NetworkManagement/interfaces/10.224.0.33/networkContainers/99e8b0e0-3524-490b-b2ca-31928e6d75a3/authenticationToken/6e4f70a6-a379-4f76-8e7a-2ae18392dc07/api-version/1",
```

Comment:
I'm curious, this is what DNC should be passing to CNS:
```
NetworkManagement/interfaces/%s/networkContainers/%s/authenticationToken/%s/api-version/1
```
but this is the "munged" version?

Reply:
@ramiro-gamarra From documentation, I believe the ordinary form of this would be `/NetworkManagement/interfaces/10.224.0.33/networkContainers/99e8b0e0-3524-490b-b2ca-31928e6d75a3?authenticationToken=6e4f70a6-a379-4f76-8e7a-2ae18392dc07&api-version=1` if it didn't have to pass through Wireserver. It's munged in that the query string is gone (replaced by slashes)... I wanted to be sure the requests would work whether those rules were applied ahead of time or not to make it more robust.